src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/client/gui/inscriptiontable/InscriptionTableScreen.java

```diff
@@ -255,28 +255,58 @@ private void sync() {
         menu.sendDataToServer(nameBar.getValue().equals(NAME_LABEL.getString()) ? null : Component.literal(nameBar.getValue()), spellGrammarArea.getAll().stream().map(e -> e.getPart().getId()).toList(), shapeGroupArea.getShapeGroupData(), compileAdditionalData());
     }
 
+    private void onPartClicked(SpellPartDraggable part, int button) {
+        ISpellPart spellPart = part.getPart();
+        if (spellPart == AMSpellParts.COLOR.get() && minecraft.options.keyUse.isActiveAndMatches(InputConstants.Type.MOUSE.getOrCreate(button))) {
```

Comment:
can we please not use keybinds in the gui

Reply:
why not? we are already on the client

Reply:
bad practice, if someone has the use key on the keyboard this code breaks

Reply:
The use key only handles in-world use of items, not GUI interactions. Minecraft doesn't support using GUIs without a mouse. See `AbstractContainerScreen#mouseClicked` (which is where I got that piece of code from).